File: java/nth-prime/src/main/java/Prime.java
import java.util.stream.IntStream;

public class Prime {
    public static boolean isPrime(int n) {
        return IntStream.rangeClosed(2, (int)Math.sqrt(n))
                .allMatch(i-> n % i != 0);
    }

    public static int nth(int n) {
        return IntStream.iterate(2, i-> i+1)
                .filter(Prime::isPrime)
                .skip(n-1)
                .findFirst()
                    .getAsInt();
    }
}

File: scala/nth-prime/src/main/scala/Prime.scala
object Prime {
  def isPrime(n: Int) =
    !(2 to math.sqrt(n).toInt).exists(n % _ == 0)

  def nth(n: Int) =
    Stream.from(2)
      .filter(isPrime)
      .drop(n-1)
      .head
}
